Provide new package numdiff

KineticTheory opened the pull request:
Numdiff can be used to compare similar files line by line and field by field, ignoring small numeric differences or/and different numeric formats. Equivalently, Numdiff is a program with the capability to appropriately compare files containing numerical fields (and not only). http://www.nongnu.org/numdiff/

Very similar to ndiff, but with additional features that we use for regression testing.

davydden commented:
`numdiff` is known to require `gettext` on `OS-X`, see https://github.com/davydden/homebrew-dealiisuite/blob/master/numdiff.rb#L13-L15  Could you pls mark it incompatible with Darwin for now with an appropriate comment. Or if you know some workaround?

KineticTheory commented:
@davydden I have added a validate method that will abort installation if the architecture is darwin.  I don't have access to a Mac so I am unable to verify if the code actually aborts.  I am only checking for darwin-x86_64.  Are there other OSX architectures I should match?

davydden commented:
@KineticTheory ah, well, nevermind. Sorry for bothering! I will update/test if after this PR and this https://github.com/LLNL/spack/pull/606 gets merged. Should be just adding a dependency to get those missing headers.

tgamblin commented:
#606 is merged.

davydden commented on var/spack/repos/builtin/packages/numdiff/package.py:
```diff
@@ -0,0 +1,56 @@
+
+    version('5.8.1',    'a295eb391f6cb1578209fc6b4f9d994e')
+
+    def validate(self, spec):
```
i think if you rebase on `develop`, then it should be something along the lines (NOT TESTED)
```
depends_on('gettext', sys.platform=='darwin')
```
with an addition of `import sys` in the head.

KineticTheory commented:
@davydden I reworked the dependency on gettext for darwin platforms and removed the abort condition.  I tested this on RHEL7.  Can you test it on OSX?

tgamblin commented:
@davydden: will merge if it works for you.

mathstuf commented:
`spack install numdiff` worked here on a 10.10 machine.
